src/app/page.tsx

```diff
@@ -1,21 +1,85 @@
-import { getTopStories } from "@/services/hn";
-import Link from "next/link";
+'use client';
+import { getTopStories } from '@/services/hn';
+import Link from 'next/link';
+import { useRef, useEffect, useState } from 'react';
 
-export default async function Home() {
-  const topstories = await getTopStories();
+export default function Home() {
+  const [start, setStart] = useState(0);
+  const limit = 10;
+  const [topstories, setTopStories] = useState([]);
+  const [message, setMessage] = useState('Läser in nyheter...');
+  const scrollTrigger = useRef(null);
+
+  useEffect(() => {
+    const fetchStories = async () => {
+      setMessage('Läser in nyheter...');
+      try {
+        const newStories = await getTopStories(start, limit);
+        if (newStories.length < limit) {
+          setMessage('Det finns inte fler nyheter att läsa in.');
+        } else {
+          setMessage('Läser in nyheter...');
+        }
+        setTopStories((prevStories) => [...prevStories, ...newStories]);
+      } catch (error) {
+        console.error('Failed to fetch stories:', error);
+        setMessage('Något gick fel vid inläsning av nyheter.');
+      }
+    };
+    fetchStories();
```

Comment:
I would like the first page to be server rendered so I don't need to wait for articles when I go to the page the first time. 

Also with this approach, using the dev server the top 10 results are duplicated since [useEffect runs twice in the dev server](https://react.dev/reference/react/useEffect#my-effect-runs-twice-when-the-component-mounts).